src/app/actions/actions.ts

```diff
@@ -0,0 +1,101 @@
+"use server";
+import { revalidatePath } from "next/cache";
+import useRequests from "@/app/service/request";
+import { ProductSchema } from "@/lib/schema";
+import { Product } from "@/lib/types";
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+const endpoints = useRequests();
+
+export async function handleServerAction(newProduct: unknown) {
+  const result1 = ProductSchema.safeParse(newProduct);
+  if (!result1.success) {
+    let errorMessage = "";
+    result1.error.issues.forEach((issue) => {
+      errorMessage = errorMessage + issue.path[0] + ": " + issue.message + ". ";
+    });
+
+    return {
+      error: errorMessage,
+    };
+  }
+  try {
+    const response = await fetch(endpoints.createProduct, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(result1.data),
+    });
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
```

Comment:
you are throwing an error inside try block only to catch it again , it might not be necessary.